pkg/tools/gen/gotypes.go

```diff
@@ -0,0 +1,85 @@
+package gen
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"strings"
+)
+
+type GoStruct struct {
+	Name          string
+	Fields        []*GoStructField
+	FieldNum      int
+	StructComment string
+}
+
+type GoStructField struct {
+	FieldName    string
+	FieldType    string
+	FieldTag     string
+	FieldTagKind string
+	FieldComment string
+}
+
+// ParseGoSourceCode parse go source code from .go file path or source code
+func ParseGoSourceCode(filename string, src interface{}) ([]*GoStruct, error) {
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, filename, src, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+	goStructList := make([]*GoStruct, 0)
+	for _, v := range f.Decls {
+		goStruct := &GoStruct{}
```

Comment:
goStruct.FieldType and goStruct.FieldName have seems never been used?

Reply:
If gen kcl schema from go struct metadata info, will use  goStruct.FieldType and goStruct.FieldName.